yt_dlp/extractor/facebook.py

```diff
@@ -889,3 +889,106 @@ def _real_extract(self, url):
         video_id = self._match_id(url)
         return self.url_result(
             f'https://m.facebook.com/watch/?v={video_id}&_rdr', FacebookIE, video_id)
+
+
+class FacebookAdsIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:[\w-]+\.)?facebook\.com/ads/library/\?id=(?P<id>\d+)'
```

Comment:
are there other subdomains we need to match besides `www`? if so, tests should be added for them

Reply:
url as follow are working but i think no specific test is needed:
https://m.facebook.com/ads/library/...
https://ja-jp.facebook.com/ads/library/...

Reply:
imo it is needed. We should at least add `only_matching` tests for them But it would be even better to add proper tests. Please at least share full URLs

Reply:
e.g. https://es-la.facebook.com/ads/library/?id=901230958115569
https://ja-jp.facebook.com/ads/library/?id=901230958115569
etc.
Those are subdomains for different webpage locales.
Should we simply add only_matching tests for them?
{
    'url': 'https://es-la.facebook.com/ads/library/?id=901230958115569',
    'only_matching': True,
}, {
    'url': 'https://m.facebook.com/ads/library/?id=901230958115569',
    'only_matching': True,
}